apps/block_scout_web/lib/block_scout_web/views/nft_helper.ex

```diff
@@ -60,6 +60,24 @@ defmodule BlockScoutWeb.NFTHelper do
     |> compose_ipfs_url()
   end
 
+  @doc """
+  Composes a full IPFS URL from the given image URL.
+
+  ## Parameters
+
+    - image_url: The URL of the image to be composed into an IPFS URL. It can be nil.
+
+  ## Returns
+
+    - A string representing the full IPFS URL or nil.
+
+  ## Examples
+
+      iex> compose_ipfs_url("QmTzQ1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1")
+      "https://ipfs.io/ipfs/QmTzQ1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1e1Y1"
+
+  """
+  @spec compose_ipfs_url(String.t() | nil) :: String.t() | nil
   def compose_ipfs_url(nil), do: nil
 
   def compose_ipfs_url(image_url) do
```

Comment:
Since this function now concern not only ipfs urls, but also arweave ones, I'd suggest the following renaming:

```suggestion
  @spec compose_resource_url(String.t() | nil) :: String.t() | nil
  def compose_resource_url(nil), do: nil

  def compose_resource_url(image_url) do
```

Reply:
Agree, since tests are also affected, I processed it in a separate commit https://github.com/blockscout/blockscout/pull/11565/commits/fda601ba868f5db10bd2b857d61a376438733a1b.